src/js/angular/models/clustermanagement/cluster.js

```diff
@@ -290,23 +291,24 @@ export class ClusterViewModel {
      * @param {Location} location - The location to add.
      */
     addToCluster(location) {
-        this._addToCluster.push(
-            new Location(
-                location.endpoint,
-                location.rpcAddress,
-                location.error,
-                location.isAvailable,
-                location.isLocal
-            )
-        );
+        this._addToCluster.push(Location.fromJSON(location));
     }
 
     /**
-     * Marks a node for deletion from the cluster.
-     * @param {Node} node - The node to delete.
+     * Marks an item for deletion from the cluster.
+     * If the item is already marked for addition, it will be removed from the addition list.
+     * Otherwise, it will be added to the deletion list.
+     *
+     * @param {Node|Location} itemToDelete - The item to delete. This can either be a Node or Location object.
+     * @return {void}
      */
-    deleteFromCluster(node) {
-        this._deleteFromCluster.push(node);
+    deleteFromCluster(itemToDelete) {
+        const locationIndex = this._addToCluster.findIndex((location) => location.endpoint === itemToDelete.endpoint);
```

Comment:
You can create some class which you can use instead of the plain js array for these addToCluster and deleteFromCluster and hide all these operations with these node lists which you do here and there

Reply:
yup